client/webserver/site/src/js/order.ts

```diff
@@ -308,14 +329,60 @@ export default class OrderPage extends BasePage {
+      Doc.setVis(!m.isCancel && (takerRedeemCoin(m) || m.active), tmpl.takerRedeem)
+      // Refunding isn't a usual part of success-path, but don't rule it out.
+      Doc.setVis(!m.isCancel && m.refund, tmpl.refund)
+    } else {
+      // Match diverged from the usual success-path, since this could have happened
+      // at any step it is hard (maybe impossible) to predict the final state this
+      // match will end up in, so show only steps that already happened plus all
+      // the possibilities on the next step ahead.
+
+      // If we don't have swap coins after revocation, we won't show the pending message.
+      Doc.setVis(!m.isCancel && (makerSwapCoin(m)), tmpl.makerSwap)
+      Doc.setVis(!m.isCancel && (takerSwapCoin(m)), tmpl.takerSwap)
+      // When match is revoked and both swaps are present, maker redeem might still show up:
+      // - as maker, we'll try to redeem until it isn't possible due to refund by taker,
+      //   so we'll have to show redeem row until we either redeem or refund
+      //   because there isn't enough data in Match m to figure out whether taker
+      //   refunded already
+      // - as taker, we should expect maker redeeming any time, so we'll have to show
+      //   redeem pending element until maker redeem shows up, or until we refund.
+      Doc.setVis(!m.isCancel && (makerRedeemCoin(m) || (takerSwapCoin(m) && m.active && !m.refund)), tmpl.makerRedeem)
+      // When maker isn't aware of taker redeem coin, once the match becomes inactive
+      // (nothing else maker is expected to do in this match) just hide taker redeem.
+      Doc.setVis(!m.isCancel && (takerRedeemCoin(m) || (makerRedeemCoin(m) && m.active && !m.refund)), tmpl.takerRedeem)
+      // As taker, show refund placeholder only if we have outstanding swap to refund.
+      // There is no need to wait for anything else, we can show refund placeholder
+      // (to inform the user that it is likely to happen) right after match revocation.
+      let expectingRefund = Boolean(takerSwapCoin(m)) // as taker
+      if (m.side === OrderUtil.Maker) {
+        // As maker, show refund placeholder only if we have outstanding swap to refund.
+        // If we don't have taker swap there is no need to wait for anything else, we
+        // can show refund placeholder (to inform the user that it is likely to happen)
+        // right after match revocation.
+        expectingRefund = Boolean(makerSwapCoin(m))
+        // If we discover taker swap we'll be trying to redeem it (instead of trying
+        // to refund our own swap) until taker refunds, so start showing refund
+        // placeholder only after taker is expected to start his refund process in
+        // this case.
+        if (takerSwapCoin(m)) {
+          const takerRefundsAfter = new Date(m.stamp + lockTimeTakerMs)
+          expectingRefund = expectingRefund && Date.now() > takerRefundsAfter.getTime()
```

Comment:
> we have no way of differentiating here whether the taker has actually refunded or if we're waiting on confirmations or wallet connectivity or something else, so this approach is maybe the best we can do with the information available?

Yes, this is my understanding too.

Reply:
> If we're the maker revoked in `TakerSwapCast`, we'll redeem up until we can refund, as long as the taker doesn't refund first. **core** does not care if the taker's locktime has passed, only whether the taker has actually refunded.

As maker, `core` won't redeem the taker's swap if it's locktime has passed. That's extremely important because maker is revealing the secret in this step, and the taker could quickly attempt to double spend their own swap contract in a refund _and_ redeem the maker's contract since they now have the secret.  Since the first seen policies are going by the wayside recently, it's even riskier.  https://github.com/decred/dcrdex/pull/1548 added this to `isRedeemable`:

https://github.com/decred/dcrdex/blob/d68ae847f8a8992ce248eeaec4c13275c3d7e6d7/client/core/trade.go#L1554-L1581

We can be sure that as maker there will be no redeem of the taker swap if the taker's swap contract has expired.

What was being displayed with the previous logic for show/hiding these fielids?

Reply:
> What was being displayed incorrectly with the previous logic for show/hiding these fielids?

**Maker**, ~2h since matched, before/after:
<img width="200" alt="image" src="https://user-images.githubusercontent.com/112318969/232386947-c922bb65-d66a-485c-ab55-627e430ce926.png"><img width="200" alt="image" src="https://user-images.githubusercontent.com/112318969/232387127-5536dd97-3920-48f2-8420-66a46275160b.png">

- so we should be waiting Maker redeeming, not refunding until Taker is able to refund right ? This is fixed here.

**Taker**, ~2h since matched, before/after:
<img width="200" alt="image" src="https://user-images.githubusercontent.com/112318969/232387881-a4ac7648-e320-481a-86ad-ff4642ae5375.png"><img width="200" alt="image" src="https://user-images.githubusercontent.com/112318969/232387950-8c1a7532-6f4a-40fc-952a-b3404fe63d08.png">

- in this PR, we display now expected refund time (if refund path gets taken)
- in this PR, we display Maker redemption (3rd line) but not Taker (our) redemption (4rd line), while on **master** we don't display 3rd line at all

Reply:
The weird thing I encountered just now is: Maker doesn't publish redeem transaction due to:
```
2023-04-17 05:32:47.103 [INF] CORE: Match 1bf6ef2071d48b6e2f9201770fbcd4605c7a5c9c75a149d5e8da82769f3c0c63 not yet redeemable: current confs = 0, required confs = 1
```
even though it already has 49 confirmations (according to **dcrdata**):
<img width="500" alt="image" src="https://user-images.githubusercontent.com/112318969/232392772-16e93164-e04c-4362-ac3b-839f99fac17e.png">

My DCR wallet is 100% synched:
<img width="500" alt="image" src="https://user-images.githubusercontent.com/112318969/232392926-98a29959-5bba-46be-a24e-ea33294a3090.png">

I think it did always publish it in the past, maybe some kind of regression ?

Reply:
Smells like another reorg issue related to "external txns" / block filters scan.  Notice the Included in Block on dcrdata shows an orphaned block.

Reply:
> Smells like another reorg issue related to "external txns" / block filters scan. Notice the Included in Block on dcrdata shows an orphaned block.

Hm, I can look into this a bit later, it has no relation to this PR I believe.

> We can be sure that as maker there will be no redeem of the taker swap if the taker's swap contract has expired.

If that's the case then there is no need to keep showing Maker redeem (3rd line) like I'm currently doing in this PR, we can hide it like that (it is also hidden on `master` but as can be seen on screenshots in my comment above it's overly aggressive and hides it regardless Taker locktime): https://github.com/decred/dcrdex/pull/2042/commits/aa25f0db8d81a129909bbd4e449a3bb22aca9c15